### PR TITLE
additional checks and minor speedup

### PR DESCRIPTION
**Pull request by vtolstov**

### CODE REVIEW

**Comment by vtolstov** [comment]
i'm pull upstream and merge it

**Comment by vtolstov** [comment]
@softwarejl what you say?

**Comment by vtolstov** [comment]
and cleanup vendor dir, all deps are fetched when building package and vendor dir only needed in final product that uses this package

**Comment by hzhou8** [comment]
@vtolstov thanks for the improvements! I appreciate that you have separate commits for different problems, but I still recommend in the future using different git branches for logically unrelated and independent changes, and sending separate PRs, which can speed up PR merging. Otherwise, one commit in the PR could block the whole PR's merging. (However, it is completely ok and recommended that multiple commits that are closely related and depended by each other go with single PR)

For this PR, the CI is failing, could you take a look?

**Comment by vtolstov** [comment]
> @vtolstov thanks for the improvements! I appreciate that you have separate commits for different problems, but I still recommend in the future using different git branches for logically unrelated and independent changes, and sending separate PRs, which can speed up PR merging. Otherwise, one commit in the PR could block the whole PR's merging. (However, it is completely ok and recommended that multiple commits that are closely related and depended by each other go with single PR)
> 
> For this PR, the CI is failing, could you take a look?

Yes, this is trivial fix, i'm already do it in qos branch, but forget to get it here.

**Comment by vtolstov** [comment]
@softwarejl thanks for suggestion, i'm try to follow it.

**Comment by vtolstov** [comment]
@softwarejl all tests passed now

File: .travis/test.sh [hunk 1]
```diff
@@ -105,7 +105,9 @@ The backup database file is sandbox/${db}2.db
 }
 
 ovn_start_db nb standalone 1 $srcdir/ovn/ovn-nb.ovsschema
-
-cd ../; go test -v
+export GO111MODULE=on
+cd ../;
```
Comment (hzhou8) [inline .travis/test.sh]: nit: ';' not needed

**Comment by vtolstov** [comment]
@softwarejl @hzhou8 i think this is fine now, can you look and merge?

File: apis.go [hunk 2]
```diff
@@ -94,23 +94,25 @@ type OVNDBApi interface {
 	Execute(cmds ...*OvnCommand) error
 
 	// Get all logical switches
-	GetLogicSwitches() []*LogicalSwitch
+	GetLogicalSwitches() ([]*LogicalSwitch, error)
```
Comment (hzhou8) [inline apis.go]: I agree that GetLogicalSwitches is better. Just to make sure @lebauce won't be surprised of this API change.

File: ovnnbimp.go [hunk 3]
```diff
@@ -88,9 +94,15 @@ func (odbi *ovnDBImp) oMapContians(s, t map[interface{}]interface{}) bool {
 }
 
 func (odbi *ovnDBImp) getRowUUIDContainsUUID(table, field, uuid string) (string, error) {
-	odbi.cachemutex.Lock()
-	defer odbi.cachemutex.Unlock()
-	for id, drows := range odbi.cache[table] {
+	odbi.cachemutex.RLock()
+	defer odbi.cachemutex.RUnlock()
+
+	cacheTable, ok := odbi.cache[table]
```
Comment (hzhou8) [inline ovnnbimp.go]: If table doesn't exist, doesn't it mean there is something fundamentally wrong? E.g. schema problem? Returning ErrorNotFound seems not appropriate for this case.

Comment (vtolstov) [reply ovnnbimp.go]: O think that i can create special error for such case

**Comment by vtolstov** [comment]
i'm rebase, and create new error that can be used to return, when table or item in table not found (useful to internal function like rowToXXX , because they used when mutex locked and uuid already getted via range iterator.

File: apis.go (hunk 2 above)
Comment (vtolstov) [reply apis.go]: i don't think that this is problem, because this is already new version, so we don't break old.

**Pull request merged**